models/user_test.go

```diff
@@ -0,0 +1,133 @@
+package models_test
+
+import (
+	_ "google-scraper/initializers"
+	"google-scraper/models"
+	. "google-scraper/tests/fabricators"
+	. "google-scraper/tests/testing_helpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("User", func() {
+	Describe("#CreateUser", func() {
+		Context("given the user with valid params", func() {
+			It("returns the user ID", func() {
+				user := &models.User{
+					Name:           "John",
+					Email:          "john@example.com",
+					HashedPassword: "secret",
+				}
+				userId, err := models.CreateUser(user)
+				if err != nil {
+					Fail("Adding user failed: " + err.Error())
+				}
+
+				Expect(userId).ToNot(BeNil())
+			})
+
+			It("returns empty error", func() {
+				user := &models.User{
+					Name:           "John",
+					Email:          "john@example.com",
+					HashedPassword: "secret",
+				}
+				_, err := models.CreateUser(user)
+
+				Expect(err).To(BeNil())
+			})
+		})
+
+		Context("given the user with INVALID params", func() {
+			Context("given the email that is already exist", func() {
+				It("returns an error", func() {
+					email := "john@example.com"
+					FabricateUser("John", email, "secret")
+
+					user := &models.User{
+						Name:           "John",
+						Email:          email,
+						HashedPassword: "secret",
+					}
+					_, err := models.CreateUser(user)
+
+					Expect(err.Error()).To(Equal(`pq: duplicate key value violates unique constraint "users_email_key"`))
```

Comment:
Is this error message will show to user? If yes I think this is not user friendly error message and I recommend to fix this in separate PR.

Reply:
It will not show this error to the user. I used the Beego validation package to validate and show formatted messages